src/com/google/javascript/jscomp/GlobalNamespace.java

```diff
@@ -1075,9 +1088,22 @@ boolean isSimpleStubDeclaration() {
     }
 
     boolean canCollapse() {
-      return !inExterns && !isGetOrSetDefinition() && (declaredType ||
+      boolean isAnnotatedNoCollapse = docInfo == null ?
+          false : docInfo.isNoCollapse();
+      if (!isAnnotatedNoCollapse) {
+        Ref ref = this.getDeclaration();
+        if (ref != null && ref.node != null) {
+          JSDocInfo info = NodeUtil.getBestJSDocInfo(ref.node);
+          if (info != null) {
+            isAnnotatedNoCollapse = info.isNoCollapse();
+          }
+        }
+      }
+
+      return !inExterns && !isGetOrSetDefinition() && !isAnnotatedNoCollapse &&
+          (declaredType ||
           (parent == null || parent.canCollapseUnannotatedChildNames()) &&
-          (globalSets > 0 || localSets > 0) &&
+          (globalSets > 0 || localSets > 0) && localSetsWithNoCollapse <= 0 &&
```

Comment:
The expression <code> localSetsWithNoCollapse <= 0 </code> can only be reached if isAnnotatedNoCollapse is false, so in this case localSetsWithNoCollapse will always be 0, right?